mm2src/coins/utxo/bchd_grpc.rs

```diff
@@ -79,11 +80,27 @@ pub struct ValidateSlpUtxosErr {
     kind: ValidateSlpUtxosErrKind,
 }
 
+impl From<ValidateSlpUtxosErr> for ValidatePaymentError {
+    fn from(err: ValidateSlpUtxosErr) -> Self {
+        match err.kind {
+            ValidateSlpUtxosErrKind::MultiReqErr(_) => Self::Transport(err.to_string()),
+            ValidateSlpUtxosErrKind::InvalidSlpTxData(_) => Self::WrongPaymentTx(err.to_string()),
```

Comment:
Shouldn't `ValidateSlpUtxosErrKind::InvalidSlpTxData(_)` map to `InvalidPaymentTxData`

Reply:
Hmm, I think it should be mapped into `InvalidPaymentTxData`

Reply:
My bad. this `test_validate_htlc_invalid_slp_utxo` confused me. It's expected to return `InvalidSlpTxData` too. thanks

Reply:
@borngraced This is not resolved yet

Reply:
No, this is correct.
We removed `InvalidPaymentTxData` so we have `TxDeserializationError `instead.
Artem suggested WrongPaymentTx [https://github.com/KomodoPlatform/atomicDEX-API/pull/1444#pullrequestreview-1108527580](https://github.com/KomodoPlatform/atomicDEX-API/pull/1444#pullrequestreview-1108527580)